### PR TITLE
Added the validation for User Creation

### PR DESCRIPTION
**Pull request by ssathiah**
Fixed the bug that created the user with an empty password.
Modified the RegisterController.java and RegisterControllerTest.java.
The fix should be in the 7.1 release.

https://cdc-nbs.atlassian.net/browse/CNDIT-694

### CODE REVIEW

File: report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java [hunk 1]
```diff
@@ -26,29 +26,35 @@ public class RegisterControllerTest {
...
-                        .param("password", "p")
+                        .param("username", "newuser123")
+                        .param("password", "password123")
                         .with(SecurityMockMvcRequestPostProcessors.jwt()))
                 .andExpect(MockMvcResultMatchers.status().isOk()).andReturn();
-        verify(registrationService).createUser(eq("u"), eq("p"));
-        Assertions.assertEquals("\"NOT_ACCEPTABLE\"", result.getResponse().getContentAsString());
+        verify(registrationService).createUser(eq("newuser123"), eq("password123"));
+        Assertions.assertEquals(USER_ALREADY_EXIST_MSG, result.getResponse().getContentAsString());
+        // Assertions.assertEquals("\"NOT_ACCEPTABLE\"", result.getResponse().getContentAsString());
```
Comment (ndduc01) [inline report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java:57]: remove these

File: report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java [hunk 2]
```diff
@@ -58,6 +64,27 @@ void createUserTestSuccessSaveReturnBadRequest() throws Exception {
                         .param("password", "")
                         .with(SecurityMockMvcRequestPostProcessors.jwt()))
                 .andExpect(MockMvcResultMatchers.status().isOk()).andReturn();
-        Assertions.assertEquals("\"BAD_REQUEST\"", result.getResponse().getContentAsString());
+        Assertions.assertEquals(USERNAME_PWD_REQ_MSG, result.getResponse().getContentAsString());
+        //Assertions.assertEquals("\"BAD_REQUEST\"", result.getResponse().getContentAsString());
```
Comment (ndduc01) [inline report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java:68]: remove these

File: report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java [hunk 3]
```diff
@@ -58,6 +64,27 @@ void createUserTestSuccessSaveReturnBadRequest() throws Exception {
...
+    }
+    @Test
+    void createUserTestSuccessUsernameMinLength() throws Exception {
+        var result = mockMvc.perform(MockMvcRequestBuilders.post("/registration")
+                        .param("username", "newuser")
+                        .param("password", "password456")
+                        .with(SecurityMockMvcRequestPostProcessors.jwt()))
+                .andExpect(MockMvcResultMatchers.status().isOk()).andReturn();
+        Assertions.assertEquals(USERNAME_PWD_MIN_LENGTH_MSG, result.getResponse().getContentAsString());
+        //Assertions.assertEquals("\"LENGTH_REQUIRED\"", result.getResponse().getContentAsString());USERNAME_PWD_MIN_LENGTH_MSG
```
Comment (ndduc01) [inline report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java:78]: remove these

File: report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java [hunk 4]
```diff
@@ -58,6 +64,27 @@ void createUserTestSuccessSaveReturnBadRequest() throws Exception {
...
+    }
+    @Test
+    void createUserTestSuccessPasswordMinLength() throws Exception {
+        var result = mockMvc.perform(MockMvcRequestBuilders.post("/registration")
+                        .param("username", "newuser456")
+                        .param("password", "pwd123")
+                        .with(SecurityMockMvcRequestPostProcessors.jwt()))
+                .andExpect(MockMvcResultMatchers.status().isOk()).andReturn();
+        Assertions.assertEquals(USERNAME_PWD_MIN_LENGTH_MSG, result.getResponse().getContentAsString());
+        //Assertions.assertEquals("\"LENGTH_REQUIRED\"", result.getResponse().getContentAsString());
```
Comment (ndduc01) [inline report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java:88]: remove these

File: report-service/src/main/java/gov/cdc/dataingestion/registration/controller/RegisterController.java [hunk 5]
```diff
@@ -11,23 +11,31 @@
...
+    private String USER_ALREADY_EXIST_MSG="User already exists.Please choose another.";
     public RegisterController(RegistrationService registrationService) {
         this.registrationService = registrationService;
     }
 
-
     @PostMapping("/registration")
-    public HttpStatus createUser(@RequestParam(value = "username") String username, @RequestParam(value = "password") String password) {
+    public String createUser(@RequestParam(value = "username") String username, @RequestParam(value = "password") String password) {
         log.info("Inside registration controller...");
```
Comment (ndduc01) [inline report-service/src/main/java/gov/cdc/dataingestion/registration/controller/RegisterController.java:24]: while you updating this, can you also update this to debug

Comment (ndduc01) [reply report-service/src/main/java/gov/cdc/dataingestion/registration/controller/RegisterController.java:24]: log.debug

File: report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java [hunk 6]
```diff
@@ -26,29 +26,35 @@ public class RegisterControllerTest {
...
+                        .param("username", "newuser123")
+                        .param("password", "password123")
                         .with(SecurityMockMvcRequestPostProcessors.jwt()))
                 .andExpect(MockMvcResultMatchers.status().isOk()).andReturn();
-        verify(registrationService).createUser(eq("u"), eq("p"));
-        Assertions.assertEquals("\"CREATED\"", result.getResponse().getContentAsString());
-
+        verify(registrationService).createUser(eq("newuser123"), eq("password123"));
+        Assertions.assertEquals(USER_CREATED_MSG, result.getResponse().getContentAsString());
+        //Assertions.assertEquals("\"CREATED\"", result.getResponse().getContentAsString());
```
Comment (ndduc01) [inline report-service/src/test/java/gov/cdc/dataingestion/registration/controller/RegisterControllerTest.java:44]: remove these

**Review by ndduc01 — COMMENTED** [review]
Some comment, otherwise looks good to me

**Comment by ndduc01** [comment]
Also, would be nice if we get rid of as much sonar queue as possible. Especially high and medium

**Comment by ssathiah** [comment]
Removed the commented code.

**Comment by ssathiah** [comment]
Modified to set a min length of six for username and eight for password.

**Review by ndduc01 — APPROVED** [review]
looks good to me

**Pull request merged**